### PR TITLE
Fix: bring back shard number suggestion

### PR DESCRIPTION
**Pull request by e-ivkov**
Bring back shard number suggestion which was accidentally removed during refactor in #675 

Also some refactoring of shard distribution struct for shard number logic to fit better.

### CODE REVIEW

File: lib/storage/src/content_manager/toc.rs [hunk 1]
```diff
@@ -190,7 +190,8 @@ impl TableOfContent {
         let CreateCollection {
             vector_size,
             distance,
-            shard_number,
+            // Can be safely ignored as `collection_shard_distribution` already has more accurate information
+            shard_number: _shard_number,
```
Comment (agourlay) [inline lib/storage/src/content_manager/toc.rs:194]: What about removing the `shard_number` argument completely?

Comment (agourlay) [reply lib/storage/src/content_manager/toc.rs:194]: This other suggestion is to add an assert making sure that the `shard_number` is reflected in the `shard_distribution` when it is defined.
So we ignore it, but we have a clear documentation and minimal harness for future invalid state.

Comment (e-ivkov) [reply lib/storage/src/content_manager/toc.rs]: After quick discussion on a call we came to conclusion that we should just assert it for consistency.

File: lib/storage/src/content_manager/toc.rs [hunk 2]
```diff
@@ -206,14 +206,20 @@ impl TableOfContent {
         let collection_path = self.create_collection_path(collection_name).await?;
         let snapshots_path = self.create_snapshots_path(collection_name).await?;
 
+        if let Some(shard_number) = shard_number {
+            assert_eq!(
```
Comment (generall) [inline lib/storage/src/content_manager/toc.rs:210]:
```suggestion
            debug_assert_eq!(
```

**Pull request merged**